From a85575f9f18136dad552a7ddc3ca1b2cfb8cd000 Mon Sep 17 00:00:00 2001
From: Andreas Scherer <andreas_github@freenet.de>
Date: Thu, 11 Jan 2018 17:08:24 +0100
Subject: [PATCH 66/66] Address issue #9 (detected on MacOS).

Do not build a shared object with conflicting external dependencies.
Unfortunately, the code base is extremely sloppy with 'external'
elements defined in 'main' modules and used in 'library' modules.
---
 Makefile | 24 +++++++++++++-----------
 1 file changed, 13 insertions(+), 11 deletions(-)

diff --git a/Makefile b/Makefile
index 1b870ba..a079c35 100644
--- a/Makefile
+++ b/Makefile
@@ -6,9 +6,7 @@
 #   In fact, CWEB 3.61 is recommended for making hardcopy or PDF documentation.
 
 #   If you prefer optimization to debugging, change -g to something like -O:
-CFLAGS = -g -fpic
-LDFLAGS = -L.
-LDLIBS = -lmmix
+CFLAGS = -g
 
 #   Uncomment the second line if you use pdftex to bypass .dvi files:
 PDFTEX = dvipdfm
@@ -64,7 +62,7 @@ TESTFILES = *.mms silly.run silly.out *.mmconfig *.mmix
 MISCFILES = Makefile makefile.dos README mmix.mp mmix.1
 ALL = $(WEBFILES) $(TESTFILES) $(MISCFILES)
 
-basic:  lib mmixal mmix
+basic:  mmixal mmix
 
 doc:    mmix-doc.ps mmixal.dvi mmix-sim.dvi
 	dvips -pp 0-13 mmixal.dvi -o mmixal-intro.ps
@@ -75,11 +73,6 @@ all:    basic mmotype mmmix
 clean:
 	rm -f *~ *.o *.c *.h *.tex *.log *.dvi *.toc *.idx *.scn *.ps core
 
-lib: libmmix.so
-
-libmmix.so: mmix-arith.o mmix-config.o mmix-io.o mmix-mem.o mmix-pipe.o
-	$(CC) -shared $^ -o $@
-
 .SECONDEXPANSION:
 mmix-pipe.o mmix-sim.o: $$(subst .o,.c,$$@)
 	sed -e "s/#define ABSTIME/& `date +%s`/" -i $<
@@ -87,8 +80,17 @@ mmix-pipe.o mmix-sim.o: $$(subst .o,.c,$$@)
 
 mmix-config.o: mmix-pipe.o
 
-mmix: mmix-sim.o lib
-	$(CC) $(LDFLAGS) $< -o $@ $(LDLIBS)
+mmmix:  mmix-arith.o mmix-pipe.o mmix-config.o mmix-mem.o mmix-io.o mmmix.c
+	$(CC) $(CFLAGS) $^ -o $@
+
+mmixal: mmix-arith.o mmixal.c
+	$(CC) $(CFLAGS) $^ -o $@
+
+mmix:   mmix-arith.o mmix-io.o mmix-sim.o
+	$(CC) $(CFLAGS) $^ -o $@
+
+mmotype: mmotype.c
+	$(CC) $(CFLAGS) $^ -o $@
 
 tarfile: $(ALL)
 	tar cvf /tmp/mmix.tar $(ALL)
-- 
2.15.1

